cmd/clusterctl/client/cluster/mover.go

```diff
@@ -60,46 +68,129 @@ func (o *objectMover) Move(namespace string, toCluster Client, dryRun bool) erro
 		log.Info("********************************************************")
 	}
 
-	objectGraph := newObjectGraph(o.fromProxy, o.fromProviderInventory)
-
 	// checks that all the required providers in place in the target cluster.
 	if !o.dryRun {
 		if err := o.checkTargetProviders(toCluster.ProviderInventory()); err != nil {
 			return errors.Wrap(err, "failed to check providers in target cluster")
 		}
 	}
 
-	// Gets all the types defines by the CRDs installed by clusterctl plus the ConfigMap/Secret core types.
+	objectGraph, err := o.getObjectGraph(namespace)
+	if err != nil {
+		return errors.Wrap(err, "failed to get object graph")
+	}
+
+	// Move the objects to the target cluster.
+	var proxy Proxy
+	if !o.dryRun {
+		proxy = toCluster.Proxy()
+	}
+
+	return o.move(objectGraph, proxy)
+}
+
+func (o *objectMover) Backup(namespace string, directory string) error {
+	log := logf.Log
+	log.Info("Performing backup...")
+
+	objectGraph, err := o.getObjectGraph(namespace)
+	if err != nil {
+		return errors.Wrap(err, "failed to get object graph")
+	}
+
+	return o.backup(objectGraph, directory)
+}
+
+func (o *objectMover) Restore(namespace string, toCluster Client, directory string) error {
+	log := logf.Log
+	log.Info("Performing restore...")
+
+	objectGraph, err := o.getObjectGraph(namespace)
```

Comment:
Hmmmm very good point. If there are existing objects in that namespace, it will fail since it will attempt to restore any objects that it finds in that object graph (including those that are already existing for that namespace). I think your suggestion to build from an empty object graph is a great idea.